kernel/src/scan/state.rs

```diff
@@ -116,14 +133,31 @@ impl<T> DataVisitor for ScanFileVisitor<'_, T> {
             // Since path column is required, use it to detect presence of an Add action
             if let Some(path) = getters[0].get_opt(row_index, "scanFile.path")? {
                 let size = getters[1].get(row_index, "scanFile.size")?;
+                let stats_json: Option<String> = getters[3].get_opt(row_index, "scanFile.stats")?;
+                let stats: Option<Stats> =
+                    stats_json.and_then(|json| match serde_json::from_str(json.as_str()) {
+                        Ok(stats) => Some(stats),
+                        Err(e) => {
+                            warn!("Invalid stats string in Add file {json}: {}", e);
+                            None
+                        }
+                    });
+
                 let dv_index = SCAN_ROW_SCHEMA
                     .index_of("deletionVector")
                     .ok_or_else(|| Error::missing_column("deletionVector"))?;
                 let deletion_vector = visit_deletion_vector_at(row_index, &getters[dv_index..])?;
                 let dv_info = DvInfo { deletion_vector };
                 let partition_values =
-                    getters[8].get(row_index, "scanFile.fileConstantValues.partitionValues")?;
-                (self.callback)(&mut self.context, path, size, dv_info, partition_values)
+                    getters[9].get(row_index, "scanFile.fileConstantValues.partitionValues")?;
```

Comment:
was this wrong before?

Reply:
No, but we've added a new `stats` field, so its index goes up.

Auto derived visitors would  be nice, because this is the kind of thing you don't remember to change and then you spend an hour debugging why it doesn't work (ask me how I know).

That's why I added to [this comment](https://github.com/delta-incubator/delta-kernel-rs/pull/227/files#diff-ba6859c8c2c33e463593ee268928c49bdeca5ed4b9699b0adf7c538732ff1229R82-R83) as well :)